### DISCUSSION TITLE
Babylon React template (version: "9.0.0" as of 2023.09.25) does not load css files as expected

### DISCUSSION THREAD

**Discussion by ivanbrizuela** [primefaces/community#115] [PrimeReact Templates] [answered]
## Problem

Using the Babylon React template (@version": "9.0.0"), unzipping and running does not work out of the box:

```shell
npm install
npm start
```

The `index.html` file can't load the any of the css files defined, reporting a 404 error on them:

Any of:

-   http://localhost:3000/assets/theme/theme-blue.css

```html
<link id="theme-css" rel="stylesheet" type="text/css" href="/assets/theme/theme-blue.css" />
```

-   http://localhost:3000/assets/layout/css/layout-blue.css

```html
<link id="layout-css" rel="stylesheet" type="text/css" href="/assets/layout/css/layout-blue.css" />
```

Results in:

<pre>
Request failed with status code 404
AxiosError: Request failed with status code 404
    at settle (http://localhost:3000/static/js/bundle.js:183363:12)
    at XMLHttpRequest.onloadend (http://localhost:3000/static/js/bundle.js:182049:66)
</pre>

## Workaround

In the `src/App.scss` import the theme and layout `scss` files:

```scss
// These imports are added
@import '../public/assets/theme/theme-blue.scss';
@import '../public/assets/layout/css/layout-blue.scss';
// This is the original import
@import './AppDemo.scss';
```
Now the *.css files are loaded, provided the other errors reported below are corrected / imports excluded.

### Inexistent folder

File `public/assets/sass/theme/_theme.scss` references an inexistent `.designer/_components` folder. If that is commented out, it works but there is no component styling.

```scss
@import './_variables';
@import '../overrides/_theme_variables';
// >> These resources don't exists:
// @import './designer/_components';
@import '../overrides/_theme_styles';
@import './vendor_extensions';
```

### Invalid path resolutions

`public/assets/sass/layout/_layout.scss` can't import the files commented out here, because they have invalid references:

```scss
@import './_variables';
@import '../overrides/_layout_variables';
// _fonts references an inexistent fonts folder
// @import '../_fonts';
@import './_mixins';
@import './_typography';
@import './_loader';
@import './_main';
@import './_topbar';
@import './_menu';
@import './_config';
@import './_footer';
@import './_utils';
@import './_dashboard';
// These three files reference images folder which are resolved incorrectly (the images do exists in the repo)
// @import './_exception';
// @import './_login';
// @import './_landing';
@import './_widgets';
@import './_help';
@import './_invoice';
@import './_wizard';
@import '../overrides/_layout_styles';
```
*NOTE:* In a previous template that I have used from PrimeReact, all *.scss files where under the src folder, no in the public folder. Maybe having all this `*.scss` in the `src` folder could prevent this errors?

## Questions

-   Am I doing something wrong?
-   How should I change the project to make it run without errors?

**Comment by habubey** [comment, marked as answer]
Thanks for the report, I will try to reproduce your code and get back to you. As my first impression, as a user, I downloaded the template, unzipped it, and ran it.
```
<link id="theme-css" rel="stylesheet" type="text/css" href="/assets/theme/theme-blue.css" />.
<link id="layout-css" rel="stylesheet" type="text/css" href="/assets/layout/css/layout-blue.css" />.
```
It works for me, I can see the theme color and component colors work well. 
Good news, the Babylon template is currently being migrated to NextJS 13 and will be available soon. Non-existing folders and wrong paths will be corrected, maybe we can update a new release without even waiting for NextJS.

**Reply by ivanbrizuela** [reply]
@habubey, indeed, it works correctly out of the box. I must have done something wrong when testing this template.

Thank you.

**Comment by ivanbrizuela** [comment]
@habubey, I found why I coudn't get this template working.

The template will work "out of the box" because all pre-compiled css files are included with the zip file.

But if the template is pushed to a private repository to continue working, future checkouts won't work due to an incorrectly configured `.gitignore` file.

I opened a specific bug report to get this corrected: #557 - Correct invalid gitignored source directory in Babylon Template (version 9.0)
